larpandoracontent/LArThreeDReco/LArTwoViewMatching/TwoViewSimpleTracksTool.cc

```diff
@@ -0,0 +1,127 @@
+       std::cout << "----> Running Algorithm Tool: " << this->GetInstanceName() << ", " << this->GetType() << std::endl;
+
+    ProtoParticleVector protoParticleVector;
+    this->FindBestTrack(overlapMatrix, protoParticleVector);
+
+    const bool particlesMade(pAlgorithm->CreateThreeDParticles(protoParticleVector));
+    return particlesMade;
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+void TwoViewSimpleTracksTool::FindBestTrack(const MatrixType &overlapMatrix, ProtoParticleVector &protoParticleVector) const
+{
+    ClusterVector sortedKeyClusters;
+    overlapMatrix.GetSortedKeyClusters(sortedKeyClusters);
+
+    for (const Cluster *const pKeyCluster : sortedKeyClusters)
+    {
+        if (!pKeyCluster->IsAvailable())
+            continue;
+
+        unsigned int n0(0), n1(0);
+        MatrixType::ElementList elementList;
+        overlapMatrix.GetConnectedElements(pKeyCluster, true, elementList, n0, n1);
+
+        if (elementList.empty())
+            continue;
+
+        MatrixType::Element bestElement(elementList.back());
+
+        for (MatrixType::ElementList::const_reverse_iterator iIter = elementList.rbegin(), iIterEnd = elementList.rend(); iIter != iIterEnd; ++iIter)
+        {
+            if(this->PassesElementCuts(iIter))
+            {
+                bestElement = *iIter;
+               if (!bestElement.GetOverlapResult().IsInitialized())
+                    continue;
+
+                if ((NULL == bestElement.GetCluster1()) || (NULL == bestElement.GetCluster2()))
+                    continue;
+
+                ProtoParticle protoParticle;
+                protoParticle.m_clusterList.push_back(bestElement.GetCluster1());
+                protoParticle.m_clusterList.push_back(bestElement.GetCluster2());
+                protoParticleVector.push_back(protoParticle);
+
+                return;
+            }
+        }
+    }
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+bool TwoViewSimpleTracksTool::PassesElementCuts(MatrixType::ElementList::const_reverse_iterator eIter) const
+{
+    if (eIter->GetOverlapResult().GetLocallyMatchedFraction() < m_minMatchedFraction)
+        return false;
+
+    if (eIter->GetOverlapResult().GetMatchingScore() < m_minMatchingScore)
+        return false;
+
+    if (eIter->GetOverlapResult().GetNMatchedReUpsampledSamplingPoints() < m_minMatchedSamplingPoints)
+        return false;
+
+    const TwoViewXOverlap &xOverlap(eIter->GetOverlapResult().GetTwoViewXOverlap());
+
+    if ((xOverlap.GetXSpan0() > std::numeric_limits<float>::epsilon()) && (xOverlap.GetXOverlapFraction0() > m_minXOverlapFraction) &&
+        (xOverlap.GetXSpan1() > std::numeric_limits<float>::epsilon()) && (xOverlap.GetXOverlapFraction1() > m_minXOverlapFraction))
```

Comment:
Similar comment for LongTracks: `xOverlap.GetXSpan0()` and `xOverlap.GetXSpan1()` can be dropped, I think.

Reply:
Completely separate comment and a purely stylistic one.
All of the above checks in this function are 'if not this then return false' which is the opposite way round to this check.
It might be neater to have this check be the same way round and have the final return be `return true`

Reply:
Done!

Reply:
I am not sure what you mean by: `xOverlap.GetXSpan0()` can be dropped?

Reply:
`xOverlap.GetXSpan0() > std::numeric_limits<float>::epsilon()` was initially in place to check that the denominator in the fraction calculation wouldn't be 0.  The new function that's now in place checks this under the hood.  So, I don't think that part of the check is needed anymore.

Reply:
Okay, nice. I will drop them, then.